web/client/api/identify.jsx

```diff
@@ -33,11 +34,11 @@ export const getFeatureInfo = (basePath, param, layer, {attachJSON, itemId = nul
     // TODO: We should move MapInfoUtils parts of the API here, with specific implementations.
     return (
         // default identify flow, valid for WMS/WMTS. It attach json data, if missing, for advanced features. TODO: make this specific by service, using layer info.
-        (attachJSON && param.info_format !== "application/json" && param.outputFormat !== "application/json")
+        (attachJSON && !isDataFormat(param))
         // add to the flow data in JSON format for highlight/zoom to feature
             ? Observable.forkJoin(
                 retrieveFlow(param),
-                retrieveFlow({ ...param, info_format: "application/json" })
+                retrieveFlow({ ...param, info_format: layer?.infoFormats?.includes(GEOJSON_MIME_TYPE) ? GEOJSON_MIME_TYPE : JSON_MIME_TYPE })
```

Comment:
can we use here `getDefaultInfoFormatValueFromLayer` ?

Reply:
We concluded that adding `getDefaultInfoFormatValueFromLayer`  instead of the manual override may lead to potential bugs in cases when `layer.featureInfo.format` carries some configuration value. This is by virtue of the implementation in `getDefaultInfoFormatValueFromLayer `.

So the solution is to leave this line as is.